core/sources/domains.move

```diff
@@ -135,7 +157,19 @@ module aptos_names::domains {
         token_helper::initialize(account);
     }
 
-    fun register_domain_generic(sign: &signer, domain_name: String, num_years: u8) acquires NameRegistryV1, RegisterNameEventsV1, SetNameAddressEventsV1 {
+    public fun init_reverse_lookup_registry_v1(account: &signer) {
+        assert!(signer::address_of(account) == @aptos_names, error::permission_denied(ENOT_AUTHORIZED));
```

Comment:
For future: This assertion is technically not necessary if we don't plan to delay enabling primary domain after we do the contract upgrade. But let's keep this for now since we've already done this in testnet. For the future, let's not do this unless necessary.